src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,75 @@
+import Poller from '../poller'
+
+jest.useFakeTimers()
+
+describe('Poller', () => {
+  let fetchFlagsFn: jest.Mock
+  let configurations: object
+  let pollInterval: number
+  let maxAttempts: number
+
+  beforeEach(() => {
+    configurations = {}
+    fetchFlagsFn = jest.fn()
+    pollInterval = 60000
+    maxAttempts = 5
+  })
+
+  it('should not start polling if it is already polling', () => {
+    const poller = new Poller(fetchFlagsFn, configurations, pollInterval, maxAttempts)
```

Comment:
Instead of maintaining global values, it can be really handy to use a function who's job it is to create the SUT (system under test) with default values and allow overrides.
```suggestion

interface PollerArgs {
  fetchFlags: () => Promise<any>
  configurations: Partial<Options>
  pollInterval: number
  maxAttempts: number
}

const getPoller = (overrides: Partial<PollerArgs> = {}): Poller => {
  const args: PollerArgs = {
    fetchFlags: jest.fn(),
    configurations: {},
    pollInterval: 60000,
    maxAttempts: 5,
    ...overrides
  }

  return new Poller(args.fetchFlags, args.configurations, args.pollInterval, args.maxAttempts)
}

describe('Poller', () => {
  it('should not start polling if it is already polling', () => {
    const poller = getPoller()
```

And so you can pass in a specific `fetchFlags` mock like this
```typescript
const fetchFlagsMock = jest.fn().mockResolvedValue(new Error('ah shit'))
const poller = getPoller({ fetchFlags: fetchFlagsMock })
```

Reply:
Beautiful - thank you!

Reply:
I'll remember this pattern for sure